functions/lib/sendgrid/create_event.py

```diff
@@ -8,92 +8,109 @@
-
-    message.dynamic_template_data = {
-      "first_name": organiser_data.get("firstName"),
-      "event_name": event_data.get("name"),
-      "event_location": event_data.get("location"),
-      "event_startDate": start_date_string,
-      "event_endDate": end_date_string,
-      "event_sport": event_data.get("sport"),
-      "event_price": event_data.get("price"),
-      "event_capacity": event_data.get("capacity"),
-      "event_isPrivate": request_data.visibility
-    }
-
-    message.template_id = CREATE_EVENT_EMAIL_TEMPLATE_ID
-
-    # TODO possibly either move this to common or make sendgrid service/ client in python
-    sg = SendGridAPIClient(SENDGRID_API_KEY)
-    response = sg.send(message)
-    
-    # Normally sendgrid return 202, reqeust in progress, but any 2XX response is acceptable.
-    if (not (response.status_code >= 200 and response.status_code < 300)):
-      raise Exception(f"Sendgrid failed to send message. e={response.body}")
-
-
-    return https_fn.Response(status=200, headers={'Access-Control-Allow-Origin', '*'})
-  except Exception as e:
-    logger.error(f"Error sending create event email. eventId={request_data.eventId}", e)
-    return https_fn.Response(status=500)
+    if not maybe_event_data.exists:
+        logger.error(
+            f"Unable to find event provided in datastore to send email. eventId={request_data.eventId}"
+        )
+        return https_fn.Response(status=400)
+
+    event_data = maybe_event_data.to_dict()
+    organiser_id = event_data.get("organiserId")
+
+    organiser_data = get_user_data(organiser_id)
+
+    try:
+        email = get_user_email(organiser_id, organiser_data)
+    except Exception as e:
+        logger.error(f"Error occured in getting organiser email.", e)
+        return https_fn.Response(status=400)
+
+    try:
+        subject = "Thank you for creating " + event_data["name"]
+        message = Mail(
+            from_email="team.sportshub@gmail.com",
+            to_emails=email,
+            subject=subject,
+        )
+        start_date: Timestamp = event_data.get("startDate").timestamp_pb()
+        end_date: Timestamp = event_data.get("endDate").timestamp_pb()
+        start_date_string = start_date.ToDatetime().strftime("%m/%d/%Y, %H:%M")
+        end_date_string = end_date.ToDatetime().strftime("%m/%d/%Y, %H:%M")
+
+        message.dynamic_template_data = {
+            "first_name": organiser_data.get("firstName"),
+            "event_name": event_data.get("name"),
+            "event_location": event_data.get("location"),
+            "event_startDate": start_date_string,
+            "event_endDate": end_date_string,
+            "event_sport": event_data.get("sport"),
+            "event_price": event_data.get("price"),
+            "event_capacity": event_data.get("capacity"),
+            "event_isPrivate": request_data.visibility,
+        }
+
+        message.template_id = CREATE_EVENT_EMAIL_TEMPLATE_ID
```

Comment:
what is this all of the unsubscribe stuff

Reply:
that was for another pr idk how it got here